sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/NonAsyncTaskShouldNotReturnNull.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using System.Threading.Tasks;
+
+namespace Tests.Diagnostics
+{
+    public class CompliantUseCases
+    {
+        public void Foo()
+        {
+            return null; // not Task
```

Comment:
Add test for arrow methods: `public Task Foo() => null;` it is less likely, but possible. Also local properties, functions and lambdas could also return null without having a `return` statement.